### PR TITLE
[DataGrid] Make SelectColumn work when using ItemsProvider

### PR DESCRIPTION
**Pull request by vnbaaij**
Some changes needed in SelectColumn logic and the DataGrid itself (by storing items and total item count in the internal gri context).

`IssueTester` page can be used to test the changes

**do not** auto-merge this. Want to clean out `IssueTester` before merging

### CODE REVIEW

**Comment by dvoituron** [comment]
I think the problem (and the solution) is not so complex :-)
The problem is due to the `Grid.ItemsProvider` which not detected.

I can create a PR with that, the code is already on my machine :-)

image

**Comment by vnbaaij** [comment]
I think with your code you can't do 'select all' when using an itemsprovider

**Comment by dvoituron** [comment]
Applying these changes on the code from the linked issue, solve the problem. What do you you want to do in addition ? The property selection is made by the "dev" code, not by the component.

**Comment by dvoituron** [comment]
Maybe to display the AllSelected icon and not the Intermediate icon only ?

File: src/Core/Components/DataGrid/FluentDataGrid.razor [hunk 1]
```diff
@@ -18,7 +18,7 @@
                           grid-template-columns=@_internalGridTemplateColumns
                           class="@GridClass()"
                           style="@Style"
-                          aria-rowcount="@(_ariaBodyRowCount + 1)"
+                          aria-rowcount="@(_internalGridContext.TotalItemCount + 1)"
```
Comment (dvoituron) [inline src/Core/Components/DataGrid/FluentDataGrid.razor:21]: Why did you add one item: `+ 1` generates an invalid value.

**Review by dvoituron — CHANGES_REQUESTED** [review]

File: src/Core/Components/DataGrid/FluentDataGrid.razor (hunk 1 above)
Comment (vnbaaij) [reply src/Core/Components/DataGrid/FluentDataGrid.razor:21]: The +1 was already there. Did not change that with this PR

**Review by dvoituron — APPROVED** [review]
